packages/design/src/DataTableNext/types.ts

```diff
@@ -0,0 +1,22 @@
+export type TableProps<T> = {
+  data: T[];
+  columns: TableColumn<T>[];
+  emptyText: string;
+  pagination?: PaginationConfig;
+  isSearchable?: boolean;
+};
+
+export type TableColumn<T> = {
+  key: keyof T & string;
+  headerText?: string;
+  onRender?: (row: T) => JSX.Element;
+  isSortable?: boolean;
+  sortType?: 'string' | 'number' | 'boolean' | 'auto';
```

Comment:
Yeah you're right, I think making the user specify what the data type they're sorting is redundant, we can infer it ourselves (and that's what we're doing with strings for example if you look at `sortAndFilter` in `useTable`). We can just remove this field and only use `sortType`.

> Additionally, I think I would expect (a, b) to have T type.

`T` type would be the whole row, but for this `onSort` we only need the specific property which we are comparing.

Reply:
Yes, I was thinking about `T` to be the whole row, but now I can't find any use case for this ;) (only creating a type for `a, b` will be easier)

Reply:
There are actually usecases for needing the whole row, this is because we sometimes need multiple pieces of data in the cell we're rendering. An example of this is the app launch button in `AppList.tsx` where we need `launchUrl`, `awsConsole`, `awsRoles`, `fqdn`, `clusterId`, and `publicAddr` just for the button

https://github.com/gravitational/webapps/blob/51fd6004c21bb7c932986d645b055b6193aa09d1/packages/teleport/src/Apps/AppList/AppList.tsx#L160-L164